src/solvers/tridiag_solver.hpp

```diff
@@ -0,0 +1,316 @@
+                 pack_rhs(b, 0, k, j, i));
+        });
+    return TaskStatus::complete;
+  }
+
+  TaskID AddTasks(TaskList &tl, TaskID dependence, const int partition, Mesh *pmesh) {
+    using namespace utils;
+    TaskID none;
+    auto partitions = pmesh->GetDefaultBlockPartitions();
+    // Should contain all fields necessary for applying the matrix to a give state vector,
+    // e.g. diffusion coefficients and diagonal, these will not be modified by the solvers
+    auto &md_base = pmesh->mesh_data.Add(container_base, partitions[partition]);
+    // Container in which the solution is stored and with which the downstream user can
+    // interact. This container only requires the fields in sol_fields
+    auto &md_u = pmesh->mesh_data.Add(container_u, partitions[partition], sol_fields);
+    // Container of the rhs, only requires fields in sol_fields
+    auto &md_rhs = pmesh->mesh_data.Add(container_rhs, partitions[partition], sol_fields);
+    // Internal solver containers
+    auto &md_100 = pmesh->mesh_data.Add(container_100, partitions[partition], sol_fields);
+    auto &md_010 = pmesh->mesh_data.Add(container_010, partitions[partition], sol_fields);
+    auto &md_001 = pmesh->mesh_data.Add(container_001, partitions[partition], sol_fields);
+
+    auto &md_100_out =
+        pmesh->mesh_data.Add(container_100_out, partitions[partition], sol_fields);
+    auto &md_010_out =
+        pmesh->mesh_data.Add(container_010_out, partitions[partition], sol_fields);
+    auto &md_001_out =
+        pmesh->mesh_data.Add(container_001_out, partitions[partition], sol_fields);
+
+    auto &md_Alo = pmesh->mesh_data.Add(container_Alo, partitions[partition], sol_fields);
+    auto &md_Adi = pmesh->mesh_data.Add(container_Adi, partitions[partition], sol_fields);
+    auto &md_Aup = pmesh->mesh_data.Add(container_Aup, partitions[partition], sol_fields);
+    auto &md_r = pmesh->mesh_data.Add(container_r, partitions[partition], sol_fields);
+
+    iter_counter = 0;
+    bool multilevel = pmesh->multilevel;
+
+    auto masks = tl.AddTask(dependence, SetMasks, md_100, md_010, md_001);
+    auto Ax100 = eqs_.Ax(tl, masks, md_base, md_100, md_100_out);
+    auto Ax010 = eqs_.Ax(tl, Ax100, md_base, md_010, md_010_out);
+    auto Ax001 = eqs_.Ax(tl, Ax010, md_base, md_001, md_001_out);
+    auto explicit_A = tl.AddTask(Ax001, SetDiagonals, md_100_out, md_010_out, md_001_out,
+                                 md_Alo, md_Adi, md_Aup);
+
+    auto copy_rhs = tl.AddTask(explicit_A, utils::CopyData<FieldTL>, md_rhs, md_r);
+    auto sol = tl.AddTask(copy_rhs, Solve, md_Alo, md_Adi, md_Aup, md_r, md_u);
+
+    sol = tl.AddTask(sol, utils::SetToZero<FieldTL>, md_r);
+    auto Ax_check = eqs_.Ax(tl, sol, md_base, md_u, md_r);
+    if (print_solution_)
+      Ax_check = tl.AddTask(Ax_check, PrintSolution, md_base, md_u, md_r, md_rhs);
+
+    return Ax_check;
+  }
+
+  Real GetSquaredResidualSum() const { return 0.0; }
+  int GetCurrentIterations() const { return 1; }
```

Comment:
This is hardcoded because it's not needed for the direct, single block solve, isn't it?